ambari-server/src/main/java/org/apache/ambari/server/controller/ServiceGroupRequest.java

```diff
@@ -57,9 +61,30 @@ public void setServiceGroupName(String serviceGroupName) {
     this.serviceGroupName = serviceGroupName;
   }
 
+  /**
+   * @return a list of associated mpack names
+   */
+  public Set<String> getMpackNames() {
+    return mpackNames;
+  }
+
+  /**
+   * @param mpackNames a list of associated mpack names
+   */
+  public void addMpackNames(Set<String> mpackNames) {
+    if (mpackNames != null) {
+      this.mpackNames.addAll(mpackNames);
+    }
+  }
+
   @Override
   public String toString() {
-    return String.format("clusterName=%s, serviceGroupName=%s", clusterName, serviceGroupName);
+    StringBuilder sb = new StringBuilder();
```

Comment:
We've been gradually moving to ToStringBuilder here instead

Reply:
Yes, we should switch this to ToStringBuilder.